scripts/create_gitlab_snapshot.py

```diff
@@ -0,0 +1,99 @@
+import datetime
+import logging
+from time import (
+    sleep,
+)
+from typing import (
+    Mapping,
+)
+
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    JSON,
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    require(config.terraform_component == 'gitlab',
+            "Select the 'gitlab' component ('dev.gitlab' or 'prod.gitlab', for example).")
+    instance = gitlab_instance_info()
+    shutdown_instance(instance)
+    create_snapshot(instance)
+
+
+def gitlab_instance_info() -> Mapping[str, JSON]:
+    filters_ = [
+        {
+            'Name': 'key-name',
+            'Values': ['azul-gitlab']
+        }
+    ]
+    return one([
+        one(reservation['Instances'])
+        for reservation in aws.ec2.describe_instances(Filters=filters_)['Reservations']
+    ])
+
+
+def shutdown_instance(instance: Mapping[str, JSON]):
+    instance_ids = [instance['InstanceId']]
+    log.info('Preparing to stop GitLab instance for %s, waiting 10 seconds '
+             'before proceeding. Hit Ctrl-C to abort',
+             config.deployment_stage)
+    sleep(10)
+    log.info('Stopping instance %s', instance_ids)
+    aws.ec2.stop_instances(InstanceIds=instance_ids)
+    log.info('Waiting for the GitLab instance to stop')
+    waiter = aws.ec2.get_waiter('instance_stopped')
+    waiter.wait(InstanceIds=instance_ids,
+                WaiterConfig=dict(MaxAttempts=9999, Delay=15))
+    log.info('Instance %s has stopped', instance_ids)
+
+
+def create_snapshot(instance: Mapping[str, JSON]):
+    # FIXME: Consolidate uses of sts.get_caller_identity
+    #        https://github.com/DataBiosphere/azul/issues/3890
+    role_id = aws.sts.get_caller_identity()['UserId'].split(':')[0]
+    resource_tags = [
+        {'Key': k, 'Value': v}
+        for k, v in {
+            'component': 'azul-gitlab',
+            'created-by': role_id,
+            'deployment': config.deployment_stage,
+            'Name': 'azul-gitlab',
+            'owner': config.owner,
+            'project': 'dcp',
+            'service': 'azul',
+        }.items()
+    ]
+    date = datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
+    response = aws.ec2.create_snapshots(Description=f'{date} Gitlab Update',
```

Comment:
We don't want a snapshot of every non-boot volume attached to the instance. 

We want a snapshot of one specific volume, the GitLab data volume, even if it is not attached to the instance.

Discover the volume by name and pass its id to https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.ServiceResource.create_snapshot. If there are more than one match raise an error.